testbed/tests/window/test_window.py

```diff
@@ -138,13 +143,191 @@ async def test_move_and_resize(main_window, main_window_probe, capsys):
         finally:
             main_window.content = orig_content
 
-    async def test_full_screen(main_window, main_window_probe):
-        """Window can be made full screen"""
-        main_window.full_screen = True
-        await main_window_probe.wait_for_window("Full screen is a no-op")
+    @pytest.mark.parametrize(
+        "initial_state, final_state",
+        [
+            # Direct switch from NORMAL:
+            (WindowState.NORMAL, WindowState.FULLSCREEN),
+            (WindowState.NORMAL, WindowState.PRESENTATION),
+            # Direct switch from FULLSCREEN:
+            (WindowState.FULLSCREEN, WindowState.NORMAL),
+            (WindowState.FULLSCREEN, WindowState.PRESENTATION),
+            # Direct switch from PRESENTATION:
+            (WindowState.PRESENTATION, WindowState.NORMAL),
+            (WindowState.PRESENTATION, WindowState.FULLSCREEN),
+        ],
+    )
+    async def test_window_state_direct_change(
+        app, initial_state, final_state, main_window, main_window_probe
+    ):
+        """Window state can be directly changed to another state."""
+        if not main_window_probe.supports_fullscreen and WindowState.FULLSCREEN in {
+            initial_state,
+            final_state,
+        }:
+            pytest.xfail("This backend doesn't support fullscreen window state.")
+        if (
+            not main_window_probe.supports_presentation
+            and WindowState.PRESENTATION
+            in {
+                initial_state,
+                final_state,
+            }
+        ):
+            pytest.xfail("This backend doesn't support presentation window state.")
+
+        try:
+            # Set to initial state
+            main_window.state = initial_state
+            await main_window_probe.wait_for_window(
+                f"Main window is in {initial_state}"
+            )
+
+            assert main_window.state == initial_state
 
-        main_window.full_screen = False
-        await main_window_probe.wait_for_window("Full screen is a no-op")
+            # Set to final state
+            main_window.state = final_state
+            await main_window_probe.wait_for_window(f"Main window is in {final_state}")
+
+            assert main_window.state == final_state
+        finally:
+            # Set to NORMAL state
+            main_window.state = WindowState.NORMAL
```

Comment:
It seems like this `main_window.state == NORMAL` and `app.current_window = main_window` cleanup tasks should be part of the window_cleanup fixture - so that every test can guarantee that it leaves the app in a clean state.